### ISSUE TITLE
AppInsight agent throws  io.netty.handler.ssl.SslHandshakeTimeoutException

### ISSUE THREAD

**Issue by ezYakaEagle442** [Azure/Azure-Spring-Apps#16]
**Describe the bug**

```console

10/27/2022, 12:43:05.546 PM
at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
10/27/2022, 12:43:05.546 PM
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
10/27/2022, 12:43:05.546 PM
at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
10/27/2022, 12:43:05.546 PM
at com.microsoft.applicationinsights.agent.internal.init.AppIdSupplier$GetAppIdTask.run(AppIdSupplier.java:139)
10/27/2022, 12:43:05.546 PM
at reactor.core.publisher.Mono.block(Mono.java:1707)
10/27/2022, 12:43:05.546 PM
at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:97)
10/27/2022, 12:43:05.546 PM
at reactor.core.Exceptions.propagate(Exceptions.java:392)
10/27/2022, 12:43:05.546 PM
reactor.core.Exceptions$ReactiveException: io.netty.handler.ssl.SslHandshakeTimeoutException: handshake timed out after 10000ms
10/27/2022, 12:43:05.546 PM
2022-10-27 10:43:05.545Z WARN c.m.a.a.i.i.AppIdSupplier$GetAppIdTask - Unable to retrieve appId: exception sending request to https://dc.services.visualstudio.com/api/profiles/9c979c8d-4655-40f4-893c-5fb29f4ff787/appId (future warnings will be aggregated and logged once every 5 minutes)
10/27/2022, 12:42:43.754 PM

```

**To Reproduce**
1. Go petclinci to ASA
2. Check the logs in Log Analytics
3. See error

**Expected behavior**
No error messages related to AppInsight should appear in the logs

**Screenshots**
N/A

**Additional context**
see also :

- [https://github.com/Azure/Azure-Spring-Apps/issues/10](https://github.com/Azure/Azure-Spring-Apps/issues/10)
- [https://github.com/netty/netty/issues/12343](https://github.com/netty/netty/issues/12343)

**Comment by hanren2021** [comment]
This should be a known AI agent issue reported here:
[Observed in debug level message "Failed to load netty-tcnative; OpenSslEngine will be unavailable"](https://github.com/microsoft/ApplicationInsights-Java/issues/2640)

This should be resolved in AI agent version 3.4.0 and later (underlying issue was [https://github.com/netty/netty/issues/12343](https://github.com/netty/netty/issues/12343))

Azure Spring App is currently using AI agent version 3.3.1, which is impacted by this issue.
We plan to upgrade the AI agent version at around 2022 Dec or 2023 Jan.

**Comment by domainname** [comment]
We have already upgraded AI agent in ASA in Jan 2023. Closing this issue now.
